gauche/kernels/string_kernels/sskkernel.py

```diff
@@ -0,0 +1,230 @@
+# GPU-compatible gpytorch version of the substring kernel implemented in https://github.com/henrymoss/BOSS/tree/master/boss/code/kernels/string
+# algorithmic description in 
+# [1] Beck, D., & Cohn, T. Learning kernels over strings using Gaussian processes. In Proceedings of the Eighth International Joint Conference on Natural Language Processing (Volume 2: Short Papers) (pp. 67-73).
+# or appendix of [2] Moss, H., Leslie, D., Beck, D., Gonzalez, J., & Rayson, P. (2020). Boss: Bayesian optimization over string spaces. Advances in neural information processing systems, 33, 15476-15486.
+
+import math
+import itertools
+import torch
+from gpytorch.kernels import Kernel
+from gpytorch.constraints import Interval
+
+def pad(s, length):
```

Comment:
Should we format function docstrings as in:

https://github.com/leojklarner/gauche/blob/main/gauche/kernels/gnn_kernels/pretrained_kernel.py

for consistency across the codebase?